casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -21,37 +21,53 @@ object Estimator {
 
   def tips[F[_]: Monad: BlockStore](
       blockDag: BlockDagRepresentation[F],
-      lastFinalizedBlockHash: BlockHash
+      genesis: BlockMessage
```

Comment:
Just an idea - we pass `genesis` block everywhere we have `BlockStore` dependency as well. Since genesis block never changes maybe it should become part of the `BlockStore` (or `BlockDagRepresentation`) in a form o a field:
```
trait BlockStore[F] {
   def genesis: F[BlockMessage]
   …
}
```
This way we could get rid of the `genesis` argument which appear in a lot of places.

Reply:
We will follow this PR up to remove the genesis block from the estimator. But in any case, this is still applicable to anything that uses the BlockStore and requires the genesis block. Thanks for the suggestion.

Reply:
I might misunderstand the issue here but my feeling is that...

If the addBlock algorithm, or, in fact, any algorithm that we run on the blockchain, cares about genesis block it will never have the right time complexity. So, if we consider this as an intermediary step that fixes a current blocker issue then it's ok. Otherwise, except for its creation, we should never care about genesis as a special block.

For this reason I wouldn't put the genesis block as a dependency but instead I would work to remove the dependencies on genesis.

Reply:
We always have to treat genesis specially because it is a block without any parents.